testing/upload_validation.ts

```diff
@@ -2140,46 +2262,79 @@ ${endpointKey ? 'endpointKey is set' : `requiresEndpointUrl is ${requiresEndpoin
     .superRefine((untypedData, context) => {
       const data = untypedData as PackVersionMetadata;
 
-      const authNetworkDomains = getAuthNetworkDomains(data);
-
-      if (!isDefined(authNetworkDomains)) {
-        // This is a Various or None auth pack.
-        return;
-      }
+      for (const authInfo of getAuthentications(data)) {
+        const {name, authentication: authentication} = authInfo;
+        const authNetworkDomains = getDeclaredAuthNetworkDomains(authentication);
 
-      // Auth network domains must match pack network domains.
-      for (const authNetworkDomain of authNetworkDomains) {
-        if (!data.networkDomains?.includes(authNetworkDomain)) {
-          context.addIssue({
-            code: z.ZodIssueCode.custom,
-            path: ['defaultAuthentication.networkDomain'],
-            message: 'The `networkDomain` in setUserAuthentication() must match a previously declared network domain.',
-          });
+        if (!isDefined(authNetworkDomains)) {
+          // This is a Various or None auth pack.
           return;
         }
+
+        const readableAuthTitle =
+          name === ReservedAuthenticationNames.Default ? 'setUserAuthentication()' : `authentication ${name}`;
+
+        // Auth network domains must match pack network domains.
+        for (const authNetworkDomain of authNetworkDomains) {
```

Comment:
What is our policy going to be for network domains when there are multiple auth types declared? Without a network domain exemption, I guess it's clear that all authentications must have exactly the same auth domain and it must be the single network domain. But if multiple network domains are allowed, is the auth network domain allowed to vary across auth types? It seems like we either need to require this or our runtime fetcher network domain validation checks need to resolve to the correct auth type to make sure the request is legal (maybe we need to do this anyway).

Reply:
I'm allowing the auth network domain to vary across auth types, since we have the manual approval gate in place. I agree we need to have tests that our fetcher enforces those various domain restrictions appropriately. cc @alan-codaio